libraries/AP_Camera/examples/tracking.py

```diff
@@ -0,0 +1,161 @@
+
+        send_gimbal_manager_pitch_yaw_angles(0, 0, 0, 0, float("NaN"), float("NaN"), math.radians(diff_y), math.radians(diff_x))
+
+#########################################################
+# Initialize the VideoCapture object
+
+# gst-launch-1.0 -v udpsrc port=5600 caps='application/x-rtp, media=(string)video, clock-rate=(int)90000, encoding-name=(string)H264' ! rtph264depay ! avdec_h264 ! videoconvert ! autovideosink sync=false
+
+gst_pipeline = (
+    "udpsrc port=5600 ! application/x-rtp,encoding-name=H264,payload=96 ! "
+    "rtph264depay ! h264parse ! queue ! avdec_h264 ! videoconvert ! appsink"
+)
+
+cap = cv2.VideoCapture(gst_pipeline, cv2.CAP_GSTREAMER)
+# cap = cv2.VideoCapture('rtsp://192.168.144.25:8554/main.264')
+
+# Load the pre-trained face detection model
+#face_cascade = cv2.CascadeClassifier(cv2.data.haarcascades + 'haarcascade_frontalface_default.xml')
+model = YOLO("yolov8n.pt") #yolo-Weights/yolov8n.pt
+
+# Define a lock for accessing the latest frame
+frame_lock = threading.Lock()
+latest_frame = None
+center_y = 0
+center_x = 0
+# Function to continuously update the latest frame
+def update_frame():
+    global latest_frame
+    while True:
+        ret, frame = cap.read()
+        if ret:
+            with frame_lock:
+                latest_frame = frame
+
+# Function to perform face detection on the latest frame
+def detect_faces():
+    global latest_frame, center_x, center_y, show_frame, frame, classf, final_list 
+    final_list = []
+    while True:
+        
+        if latest_frame is not None:
+            with frame_lock:
+                frame = latest_frame
+            gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+            # Your face detection code here
+            # For example:
+
+            results = model.track(frame,stream=True)
+            for r in results:
+                boxes = r.boxes
+                for box in boxes:
+                    #bounding box
+                    classf = int(box.cls)
+                    name = model.names[classf]
+                    print("classification is:", name)
+                    if (name != "person" or box.conf < 0.7):
+                        continue
+                    x1, y1, x2, y2 = box.xyxy[0]
+                    x1, y1, x2, y2 = int(x1), int(y1), int(x2), int(y2) # convert to int values
+
+                    # height, width = img.shape[:2]
+
+                    #Calculate the center coordinates
+                    # img_center_x = width // 2
+                    # img_center_y = height // 2
+                    
+                    #put box in cam
+                    
+                    final_list = (x1, y1, x2, y2)
+                    cv2.rectangle(frame, (x1, y1), (x2, y2), (255, 0, 255), 3)
+
+            # faces = face_cascade.detectMultiScale(gray, scaleFactor=1.3, minNeighbors=5, minSize=(30, 30))
+            faces= [final_list]
+            if len(faces) == 0:
+                # Do something if no faces were detected
```

Comment:
We should be sending the CAMERA_TRACKING_IMAGE_STATUS message at regular intervals (1hz?) so that AP knows if we can't track the object any more

Reply:
We should support [MAV_CMD_CAMERA_STOP_TRACKING](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L1806)